core/io/file_access_zip.cpp

```diff
@@ -147,9 +147,14 @@ unzFile ZipArchive::get_file_handle(String p_file) const {
 	return pkg;
 }
 
-bool ZipArchive::try_open_pack(const String &p_path, bool p_replace_files) {
+bool ZipArchive::try_open_pack(const String &p_path, bool p_replace_files, size_t p_offset = 0) {
 	//printf("opening zip pack %ls, %i, %i\n", p_name.c_str(), p_name.extension().nocasecmp_to("zip"), p_name.extension().nocasecmp_to("pcz"));
-	if (p_path.get_extension().nocasecmp_to("zip") != 0 && p_path.get_extension().nocasecmp_to("pcz") != 0) {
+    // load with offset feature only supported for PCK files
+    if (p_offset != 0) {
```

Comment:
If this is a developer usage error, should this print an error message or anything?

Reply:
Added error logging using [ERR_FAIL_COND_V_MSG()](https://github.com/godotengine/godot/blob/7e38e8fdb0c05f0216e4322806315a025dc05b6f/core/error_macros.h#L383).
If load zip file with offset is called, the debugger will log an error:
<img width="547" alt="Screen Shot 2020-07-14 at 2 16 33 PM" src="https://user-images.githubusercontent.com/14120761/87467149-1771ce00-c5dd-11ea-9cc5-6bb0d7e13026.png">

Reply:
Updated the error message to: Load with offset feature not supported for Zip files.

Reply:
Thanks!

Reply:
It looks like [file_access_pack.cpp](https://github.com/godotengine/godot/blob/c2151e18135817c9f926a5a00341016ac77301d4/core/io/file_access_pack.cpp#L39) calls try_open_pack for each of it's sources. Since [sources is a vector of PackSources](https://github.com/godotengine/godot/blob/c2151e18135817c9f926a5a00341016ac77301d4/core/io/file_access_pack.h#L94), it might be possible to call try_open_pack with an offset on a zip pack without meaning to. Would this error prevent a user with zip packs from using PAD?

Reply:
As discussed offline we'll keep the error as is for now and see what the Godot maintainers think.